control-plane/csi-driver/src/bin/node/main_.rs

```diff
@@ -309,6 +319,31 @@ pub(super) async fn main() -> anyhow::Result<()> {
         check_ana_and_label_node(&kube_client, node_name, nvme_enabled).await?;
     }
 
+    let conn_fallback = matches.get_flag("nvme-connect-fallback");
+    // Check and store if this node has an RDMA device so that nvme connect over RDMA
+    // can be done. If this node is also an io-engine node with rdma enabled and working,
+    // then this check will always set capability true.
+    let ibv_output = tokio::process::Command::new("ibv_devinfo")
+        .arg("-l")
+        .output()
+        .await;
+
+    if ibv_output.as_ref().is_ok_and(|s| {
+        !String::from_utf8(s.stdout.clone())
+            .unwrap_or_default()
+            .starts_with("0 HCA")
+    }) {
+        info!(
+            "host node rdma capable. conn_fallback({conn_fallback:?}), {:?}",
+            String::from_utf8(ibv_output.unwrap().stdout.clone())
```

Comment:
reuse the output from above rather than rebuild it here

Reply:
> reuse the output from above rather than rebuild it here

Directly using may be tricky (unless we rearrange using some if else) as that's inside closure along with `status` check. I've removed the redundant `clone()` now, and `String::from_utf8` doesn't do any cloning/copying of vector internally.

Reply:
I would say split it into multiple statements.
Also I think there's a bug, if string parsing fails, then it's being taken as the host is rdma capable?

Maybe something like this:
```suggestion
    let output = ibv.output.as_ref().flat_map(|s| String::from_utf8(s.stdout.clone().ok());
    match output {
      Some(output) if ibv.status.success() && !output.starts_with("0 HCA") => {
        info!(
            "host node rdma capable. conn_fallback({conn_fallback:?}), {:?}",
            String::from_utf8(ibv_output.unwrap().stdout)
        );
        // todo: check for the presence of nvme_rdma too. But don't bail out as we do for nvme_tcp.
        let _ = RDMA_CONNECT_CHECK.set((conn_fallback, true));
      _ => {
        error!("Error executing ibv_devinfo, or no output received. conn_fallback({conn_fallback:?}), {ibv_output:?}");
        let _ = RDMA_CONNECT_CHECK.set((conn_fallback, false));
      }
    }
```

Reply:
ok so I've rearranged to keep it simple now, no fancy combinators. 
> Also I think there's a bug, if string parsing fails, then it's being taken as the host is rdma capable?

Handled this by erring towards incapability.